kolibri/core/assets/src/views/side-nav.vue

```diff
@@ -216,36 +153,111 @@
       },
     },
     methods: {
-      navigate(option) {
-        if (option.href) {
-          window.location.href = option.href;
-        } else if (option.action) {
-          option.action();
-        }
-      },
       toggleNav() {
         this.$emit('toggleSideNav');
       },
-      pageIsActive(pageName) {
-        return this.topLevelPageName === pageName;
+      setPageIsActive(navItem) {
+        navItem.active = window.location.pathname.startsWith(navItem.url);
+        return navItem;
+      },
+      compareMenuItems(navItemA, navItemB) {
+        // Compare menu items to allow sorting by the following priority:
+        // First sort by Section, if two items have different section numbers
+        // the lowest number is first
+        if (navItemA.section !== navItemB.section) {
+          return navItemA.section - navItemB.section;
+        }
+        // Next sort by role
+        // Nav items with no roles will be placed first
+        // as index will be -1
+        if (navItemA.role !== navItemB.role) {
+          return navItemRoleOrder.indexOf(navItemA.role) - navItemRoleOrder.indexOf(navItemB.role);
+        }
+        // Next sort by priority
+        if (navItemA.priority !== navItemB.priority) {
+          return navItemA.priority - navItemB.priority;
+        }
+        // If all else fails, sort by alphabetical order of URL
+        // to give consistent ordering of nav bar items across languages
+        if (navItemA.url.toUpperCase() < navItemB.url.toUpperCase()) {
+          return -1;
+        }
+        if (navItemA.url.toUpperCase() > navItemB.url.toUpperCase()) {
+          return 1;
+        }
+        // Still no difference?
+        // There is no difference!
+        return 0;
+      },
+      filterByRole(navItem) {
+        if (!navItem.role) {
+          // No role defined, so always show
+          return true;
+        }
+        if (navItem.role === UserKinds.COACH) {
+          return this.isCoach || this.isAdmin || this.isSuperuser;
+        }
+        if (navItem.role === UserKinds.ADMIN) {
+          return this.isAdmin || this.isSuperuser;
+        }
+        if (navItem.role === UserKinds.CAN_MANAGE_CONTENT) {
+          return this.canManageContent || this.isSuperuser;
+        }
+        if (navItem.role === UserKinds.SUPERUSER) {
+          return this.isSuperuser;
+        }
+        if (navItem.role === UserKinds.ANONYMOUS) {
+          return !this.isUserLoggedIn;
+        }
+        if (navItem.role === UserKinds.LEARNER) {
+          return this.isLearner || this.isCoach || this.isAdmin || this.isSuperuser;
+        }
+      },
+      addDividers(itemList) {
```

Comment:
Yeah, it would need additional documentation and cleanup.

Reply:
a way to gauge ... If a developer is interested in adding or removing dividers based on mockups from designers, will it be straightforward to know how to implement the mockups

Reply:
Well, part of this design (by necessity) is that we no longer have direct control over the exact positioning of dividers. In this implementation, they are added at boundaries between sections (so, when section 0 ends, and section 1 begins). There is no explicit control over this, in order to allow for dynamic expansion of the side nav. This may well be undesirable, but unless we follow the build time configuration approach you suggest, we cannot guarantee the contents of the side nav completely.